sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -275,10 +275,17 @@ Page<DataFeed> NextPageFunc(string nextLink, int? pageSizeHint)
         /// A <see cref="Response{T}"/> containing the result of the operation. The result is a <c>string</c>
         /// containing the ID of the newly created feed.
         /// </returns>
-        /// <exception cref="ArgumentNullException"><paramref name="dataFeed"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="dataFeed"/>, <paramref name="dataFeed"/>.Name, <paramref name="dataFeed"/>.DataSource, <paramref name="dataFeed"/>.Granularity, <paramref name="dataFeed"/>.Schema, <paramref name="dataFeed"/>.IngestionSettings, or <paramref name="dataFeed"/>.IngestionSettings.IngestionStartTime is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="dataFeed"/>.Name is empty.</exception>
         public virtual async Task<Response<string>> CreateDataFeedAsync(DataFeed dataFeed, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNull(dataFeed, nameof(dataFeed));
+            Argument.AssertNotNullOrEmpty(dataFeed.Name, $"{nameof(dataFeed)}.{nameof(dataFeed.Name)}");
+            Argument.AssertNotNull(dataFeed.DataSource, $"{nameof(dataFeed)}.{nameof(dataFeed.DataSource)}");
+            Argument.AssertNotNull(dataFeed.Granularity, $"{nameof(dataFeed)}.{nameof(dataFeed.Granularity)}");
+            Argument.AssertNotNull(dataFeed.Schema, $"{nameof(dataFeed)}.{nameof(dataFeed.Schema)}");
+            Argument.AssertNotNull(dataFeed.IngestionSettings, $"{nameof(dataFeed)}.{nameof(dataFeed.IngestionSettings)}");
+            Argument.AssertNotNull(dataFeed.IngestionSettings.IngestionStartTime, $"{nameof(dataFeed)}.{nameof(dataFeed.IngestionSettings)}.{nameof(dataFeed.IngestionSettings.IngestionStartTime)}");
```

Comment:
Since we don't have a constructor in `DataFeed` anymore, moved the validation to the service method instead.

Reply:
Is this an example of [service parameter validation](https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation)?

Reply:
Did we also consider creating a ctor on `DataFeed` that takes only the required property values?

Reply:
> 
> 
> Is this an example of [service parameter validation](https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation)?

Possibly. There are a few points to consider, though:
- The service parameter validation guideline has always been a bit nebulous to me. In fact, we don't usually do range validation, but we see a lot of null/empty checks in our clients to ensure the user is not passing an invalid value by mistake. We usually put these in the constructors, but that's something we can't really do after this PR. What's the practical difference between adding these checks to the constructor and adding them to the service methods?
- Except for the empty string check, I'd say these validations are not really used to validate the parameter values, but to make sure they're present and follow the request format specified by the swagger (which mark them as required and not nullable).

> 
>
> Did we also consider creating a ctor on DataFeed that takes only the required property values?

One piece of feedback we got during archboard review was that we should avoid constructors with complex/client types (https://github.com/Azure/azure-sdk-for-net/issues/16321), so we're making the `DataFeed` constructor parameterless. If you're talking about creating a constructor that takes only the required BCL types (having a `ingestionStartTime` `DateTimeOffset` instead of a whole `DataFeedIngestionSettings` object), we didn't really think about it, but:
- We'll still need the parameterless constructor for the Update method. One flaw in the current design is that users need to instantiate a model with the parameters required only during creation even when they just want to update it.
- It may not be feasible to do it in `DataFeed` because of the 13 different possible set of parameters a `DataFeedSource` has.

Reply:
what about having `DataFeed()` and `DataFeed(with all things needed)`, so the user can decide which on to use, but if they need help figuring out what is mandatory, then the use the complex constructor

Reply:
> what about having `DataFeed()` and `DataFeed(with all things needed)`, so the user can decide which on to use, but if they need help figuring out what is mandatory, then the use the complex constructor

Since this would be an additive change, I would prefer to confirm whether this is a real concern to users first.